test/unit/iframe.js

```diff
@@ -1,94 +1,500 @@
+        iframe.document.createElement = createElementStub;
+        iframe.appendStyleTag();
+        assert.calledOnce(styleTagAppendChildStub);
+        assert.calledWith(styleTagAppendChildStub, textNode);
+        assert.calledOnce(createTextNodeStub);
+        assert.calledWith(createTextNodeStub, iframe.css);
+      });
+    });
+
+    describe('getters', () => {
+      describe('width', () => {
+        it('returns max width of parent', () => {
+          assert.equal(iframe.width, iframe.parent.style['max-width']);
+        });
+      });
+
+      describe('document', () => {
+        it('returns element\'s content window\'s document if the content window body exists', () => {
+          iframe.el = {contentWindow: {document: {body: {}}}};
+          assert.equal(iframe.document, iframe.el.contentWindow.document);
+        });
+
+        it('returns element\'s document if the document body exists', () => {
+          iframe.el = {document: {body: {}}};
+          assert.equal(iframe.document, iframe.el.document);
+        });
+
+        it('returns element\'s content document if the content document body exists', () => {
+          iframe.el = {contentDocument: {body: {}}};
+          assert.equal(iframe.document, iframe.el.contentDocument);
+        });
+
+        it('returns undefined if there is no content window document body, document body, or content document body', () => {
+          iframe.el = {contentWindow: null, document: null, contentDocument: null};
+          assert.isUndefined(iframe.document);
+        });
+      });
+
+      describe('conditionalCSS', () => {
+        it('returns empty string if browser features include transition, transform, and animation', () => {
+          iframe.browserFeatures = {
+            transition: true,
+            transform: true,
+            animation: true,
+          };
+          assert.equal(iframe.conditionalCSS, '');
+        });
+
+        it('returns conditionalStyles if browser features does not include transition', () => {
+          iframe.browserFeatures.transition = false;
+          assert.equal(iframe.conditionalCSS, conditionalStyles);
+        });
+
+        it('returns conditionalStyles if browser features does not include transform', () => {
+          iframe.browserFeatures.transform = false;
+          assert.equal(iframe.conditionalCSS, conditionalStyles);
+        });
+
+        it('returns conditionalStyles if browser features does not include animation', () => {
+          iframe.browserFeatures.animation = false;
+          assert.equal(iframe.conditionalCSS, conditionalStyles);
+        });
+      });
+
+      describe('css', () => {
+        it('returns properly formatted CSS', () => {
+          const customCSS = '@media (max-width: 100px) { .product { background: blue; }  } .product:hover .btn { background: orange; } .btn:hover { color: green; } .btn { color: red; }';
+          assert.include(iframe.css, iframe.stylesheet);
```

Comment:
We should set `iframe.css` to a `const` so we don't need to call it multiple times